src/rust/engine/fs/store/src/lib.rs

```diff
@@ -1440,7 +1442,16 @@ impl Store {
     destination: PathBuf,
     target: String,
   ) -> Result<(), StoreError> {
-    hard_link(target, destination)?;
+    // On macOS, copy uses a copy-on-write syscall (fclonefileat) which creates a disconnected
+    // clone. It is more defensive than a hardlink, but has the same requirement that the source
+    // and destination filesystem are the same.
+    //
+    // It also has the benefit of playing nicely with Docker for macOS file virtualization: see
+    // #18162.
+    #[cfg(target_os = "macos")]
+    copy(target, destination).await?;
+    #[cfg(target_os = "linux")]
```

Comment:
This would be better as "not macos". Think windows.

Reply:
Good call.